library/blivet.py

```diff
@@ -559,6 +591,49 @@ def _create(self):
             label = get_format("disklabel", device=self._device.path, label_type=disklabel_type)
             self._blivet.format_device(self._device, label)
 
+class BlivetMDRaidPool(object):
+
+    def __init__(self, blivet_obj, pool, disks):
+        self._blivet = blivet_obj
+        self._pool = pool
+        self._disks = disks
+
+    def create(self):
+        raid_name = "mdraid-%s" % self._pool['name']
+        member_names = [x.name for x in self._disks]
+        raid_level = self._pool['raid_level']
+
+        # check validity of argument values
+
+        # valid RAID level (e.g. "raid0", "raid1", "linear"...)
+        try:
+            raid_level_obj = get_raid_level(raid_level)
+        except RaidError:
+            raise BlivetAnsibleError("invalid RAID level value '%s'" % raid_level)
+
+        # valid member count
+        if raid_level_obj.min_members < len(member_names):
+            raise BlivetAnsibleError("at least %s members are required for RAID level '%s'" % (raid_level_obj.min_members, raid_level))
+
+        if safe_mode:
+            raise BlivetAnsibleError("cannot create new RAID '%s' in safe mode" % safe_mode)
+        raid_members = create_raid_members(self._blivet, member_names)
+
+        raid_array = self._blivet.new_mdarray(name=raid_name, level=raid_level,
+                                               member_devices=len(raid_members),
+                                               total_devices=(len(raid_members)),
+                                               parents=raid_members)
+
+
+        if not safe_mode:
+            self._blivet.create_device(raid_array)
+
+            try:
+                do_partitioning(self._blivet)
+            except Exception:
+                raise BlivetAnsibleError("failed to allocation partitions for pool '%s'" % self._pool['name'])
```

Comment:
I don't think this class serves any purpose as it is now. The only thing I think it adds is the `new_mdarray` call, which could easily go into `BlivetPool._create_members`.